syntax = "proto3";

package tensorboard_plugin_geo;

// A GeoPluginData encapsulates information on which plugins are able to make
// use of a certain summary value.
message GeoPluginData {
  enum ContentType {
    UNDEFINED = 0;
    VERTICES = 1;
    FACES = 2;  // Triangle face.
    FEATURES = 3;
    VERT_COLORS= 4;
    FEAT_COLORS= 5;
  }

  // Version `0` is the only supported version.
  int32 version = 1;

  // The name of the mesh summary this particular summary belongs to.
  string name = 2;

  // Type of data in the summary.
  ContentType content_type = 3;

  // JSON-serialized dictionary of ThreeJS classes configuration.
  string json_config = 5;

  // Shape of underlying data. Cache it here for performance reasons.
  repeated int32 shape = 6;

  // Bitmask of all existing components in addition current component.
  uint32 components = 7;
}